mesonbuild/modules/pkgconfig.py

```diff
@@ -701,6 +701,9 @@ def parse_variable_list(vardict: T.Dict[str, str]) -> T.List[T.Tuple[str, str]]:
             if mesonlib.is_freebsd():
                 pkgroot = os.path.join(_as_str(state.environment.coredata.get_option(mesonlib.OptionKey('prefix'))), 'libdata', 'pkgconfig')
                 pkgroot_name = os.path.join('{prefix}', 'libdata', 'pkgconfig')
+            if mesonlib.is_haiku():
+                pkgroot = os.path.join(_as_str(state.environment.coredata.get_option(mesonlib.OptionKey('prefix'))), 'develop/lib/pkgconfig')
```

Comment:
Perhaps use the same style as like 702 where each path component is its own argument.